packages/flutter_tools/lib/src/macos/swift_packages.dart

```diff
@@ -0,0 +1,402 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/file_system.dart';
+import '../base/template.dart';
+import '../base/version.dart';
+
+const String minimumSwiftToolchainVersion = '5.9';
+
+const String _swiftPackageTemplate = '''
+// swift-tools-version: {{swiftToolsVersion}}
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+//
+//  Generated file. Do not edit.
+//
+
+import PackageDescription
+
+let package = Package(
+    name: "{{packageName}}",
+    {{#platforms}}
+    platforms: [
+        {{platforms}}
+    ],
+    {{/platforms}}
+    products: [
+        {{products}}
+    ],
+    dependencies: [
+        {{dependencies}}
+    ],
+    targets: [
+        {{targets}}
+    ]
+)
+''';
```

Comment:
No this file will not really be editable by the user since it's generated. Unfortunately, swift packages don't have as much control as CocoaPods did. 

So `use_frameworks!` would tell CocoaPods to use dynamic libraries rather than static - that's not possible with SPM. You can [define whether the package itself](https://developer.apple.com/documentation/packagedescription/product/library(name:type:targets:)#parameters) is static or dynamic, but not dependencies.

Also, FYI, SPM recommends not setting what type and letting Xcode decide. It seems to default to static. 

I did confirm that privacy manifest are still included in the app even if statically linked. SPM puts them in a `.bundle` in the Runner.app directory. I also tested that Xcode's Generate Report archival process found them.